Shared/Samples/Show viewshed from geoelement in scene/ShowViewshedFromGeoelementInSceneView.swift

```diff
@@ -0,0 +1,197 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct ShowViewshedFromGeoelementInSceneView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The timer for the moving tank animation.
+    @State private var animationTimer: Timer!
+    
+    var body: some View {
+        SceneView(
+            scene: model.scene,
+            cameraController: model.cameraController,
+            graphicsOverlays: [model.graphicsOverlay],
+            analysisOverlays: [model.analysisOverlay]
+        )
+        .onSingleTapGesture { _, mapPoint in
+            // Start a timer to animate the tank moving towards the new waypoint.
+            model.waypoint = mapPoint
+            animationTimer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { _ in
```

Comment:
I still feel there is a potential bug here. If you tap on the scene view several times, each time the state property "animationTimer" will be pointing to the new timer instance, and we lose the reference to the old timer.

---

Also, I feel it a bit strange to have a timer as a state property, while the only purpose is to make it mutable. Is it possible to move the whole animation logic to the model, and only pass the waypoint to that method? Please hold for Phil's thoughts here.

Reply:
I agree with @yo1995. The animation method being called is on the model. The waypoint being navigated to is on the model. The timer itself should be on the model. On tap, all the view should need to do is call a method on the model like `func move(toWaypoint waypoint: Point)`. There should also be a `func stopMoving()` method that is called by the view on disappear to stop the timer when the view goes away.

In addition, by having it on the model, the timer could be invalidated automatically when the waypoint becomes `nil` using a property observer of the `waypoint` property.